src/test/java/org/jenkinsci/plugins/workflow/graph/FlowNodeTest.java

```diff
@@ -121,4 +124,251 @@ private static void assertActiveSteps(WorkflowRun b, String... expected) {
         assertEquals(Arrays.asList(expected), actual);
     }
 
+    @Issue("JENKINS-27395")
+    @Test
+    public void enclosingBlocks() throws Exception {
```

Comment:
I suspect this is giving good coverage, but it really needs to be split up into several smaller testcases to let us debug any issues introduced in a specific case (for example: incomplete parallels).

Reply:
awwwww. Can you give me some examples of dummy Jenkinsfiles that you want covered?

Reply:
Hold on, let me review other parts and then will explain more